### PR TITLE
Split request node certificate

### PR DESCRIPTION
**Pull request by jcbsmpsn**
Split the `RequestNodeCertificate` function so the behavior can also be used by non-node callers.

### CODE REVIEW

**Comment by jcbsmpsn** [comment]
cc @mikedanese

File: pkg/kubelet/util/csr/csr_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	}
+	csr := certificates.CertificateSigningRequest{}
+	csr.Name = "fake-certificate-signing-request-name"
+	csr.UID = "fake-uid"
+	return &csr, nil
+}
+
+func (c *fakeClient) Watch(opts v1.ListOptions) (watch.Interface, error) {
+	c.watch = watch.NewFakeWithChanSize(1, false)
+	time.Sleep(1 * time.Second)
```
Comment (mikedanese) [inline pkg/kubelet/util/csr/csr_test.go]: why is this needed?

Comment (jcbsmpsn) [inline pkg/kubelet/util/csr/csr_test.go]: Oops. Totally unnecessary.

File: pkg/kubelet/util/csr/csr_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	}
+}
+
+func TestRequestNodeCertificateErrorCreatingCSR(t *testing.T) {
+	client := &fakeClient{
+		failureType: createError,
+	}
+	privateKeyData, err := certutil.MakeEllipticPrivateKeyPEM()
+	if err != nil {
+		t.Fatalf("unable to generate a new private key: %v", err)
```
Comment (mikedanese) [inline pkg/kubelet/util/csr/csr_test.go]: capitalize unable.

File: pkg/kubelet/util/csr/csr_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	}
+	if certData != nil {
+		t.Errorf("Got cert data, wanted nothing as there should have been an error.")
+	}
+}
+
+func TestRequestNodeCertificate(t *testing.T) {
+	privateKeyData, err := certutil.MakeEllipticPrivateKeyPEM()
+	if err != nil {
+		t.Fatalf("unable to generate a new private key: %v", err)
```
Comment (mikedanese) [inline pkg/kubelet/util/csr/csr_test.go]: same,

File: pkg/kubelet/util/csr/csr_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+	}
+
+	certData, err := RequestNodeCertificate(&fakeClient{}, privateKeyData, "fake-node-name")
+	if err != nil {
+		t.Errorf("Got %v, wanted no error.")
```
Comment (mikedanese) [inline pkg/kubelet/util/csr/csr_test.go]: needs an argument

File: pkg/kubelet/util/csr/csr_test.go [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+	failureType FailureType
+}
+
+func (c *fakeClient) Create(*certificates.CertificateSigningRequest) (*certificates.CertificateSigningRequest, error) {
+	if c.failureType == createError {
+		return nil, fmt.Errorf("fakeClient failed creating request")
+	}
+	csr := certificates.CertificateSigningRequest{}
+	csr.Name = "fake-certificate-signing-request-name"
+	csr.UID = "fake-uid"
```
Comment (mikedanese) [inline pkg/kubelet/util/csr/csr_test.go]: set these in the struct decleration?

**Comment by mikedanese** [comment]
Looks good. Thanks.

File: pkg/kubelet/util/csr/csr_test.go (hunk 5 above)
Comment (jcbsmpsn) [inline pkg/kubelet/util/csr/csr_test.go]: Sadly, no. These properties are not in the struct but are inherited from an included struct.

Comment (jcbsmpsn) [inline pkg/kubelet/util/csr/csr_test.go]: And a nested struct (as far as I can tell) is difficult to initialize directly.

File: pkg/kubelet/util/csr/csr_test.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+
+	csr := certificates.CertificateSigningRequest{
+		Status: certificates.CertificateSigningRequestStatus{
+			Conditions: []certificates.CertificateSigningRequestCondition{
+				condition,
+			},
+			Certificate: []byte{},
+		},
+	}
+	csr.UID = "fake-uid"
```
Comment (jcbsmpsn) [inline pkg/kubelet/util/csr/csr_test.go]: Comes from a nested struct, it's difficult to initialize directly.

**Comment by jcbsmpsn** [comment]
cc @dchen1107

File: pkg/kubelet/util/csr/csr_test.go (hunk 5 above)
Comment (mikedanese) [inline pkg/kubelet/util/csr/csr_test.go]:
```golang
certificates.CertificateSigningRequest{
    ObjectMeta: metav1.ObjectMeta{
        Name: "fake-certificate-signing-request-name",
        UID: "fake-uid",
    },
}
```

**Comment by jcbsmpsn** [comment]
@k8s-bot node e2e test this

**Comment by jcbsmpsn** [comment]
@k8s-bot cri node e2e test this

**Comment by jcbsmpsn** [comment]
@k8s-bot cri node e2e test this

**Comment by mikedanese** [comment]
@Random-Liu @vishh @timstclair can I get an approve?

**Comment by jcbsmpsn** [comment]
@k8s-bot gci gce e2e test this

**Comment by jcbsmpsn** [comment]
@k8s-bot gci gce e2e test this

**Comment by mikedanese** [comment]
@kubernetes/sig-node-api-reviews can someone please approve this change?

**Comment by mikedanese** [comment]
@yujuhong thanks!

**Comment by jcbsmpsn** [comment]
@k8s-bot non-cri node e2e test this

**Comment by jcbsmpsn** [comment]
@k8s-bot cvm gke e2e test this

**Comment by jcbsmpsn** [comment]
@k8s-bot gce etcd3 e2e test this
